src/transformers/models/wav2vec2/feature_extraction_wav2vec2.py

```diff
@@ -181,9 +181,13 @@ def __call__(
                 "Failing to do so can result in silent errors that might be hard to debug."
             )
 
-        is_batched = bool(
-            isinstance(raw_speech, (list, tuple))
-            and (isinstance(raw_speech[0], np.ndarray) or isinstance(raw_speech[0], (tuple, list)))
+        is_batched_numpy = isinstance(raw_speech, np.ndarray) and len(raw_speech.shape) == 2
+        if is_batched_numpy:
+            assert (
+                len(raw_speech.shape) == 2
```

Comment:
Note that `is_batched_numpy` will **only** be True if `raw_speech` is 2-d based on the above line. So this assert will never be triggered! Probably what we can do is set `is_batched_numpy` based on whether `raw_speech` is a numpy array with dimensionality 2 or more:
```python
is_batched_numpy = isinstance(raw_speech, np.ndarray) and len(raw_speech.shape) > 1
```
And then check to make sure it has the correct dims:
```python
if is_batched_numpy:
    if len(raw_speech.shape) > 2:
        raise ValueError(f"Only mono-channel audio is supported for input to {self}"
```
(we usually go with ValueError's over asserts in transformers 🤗)